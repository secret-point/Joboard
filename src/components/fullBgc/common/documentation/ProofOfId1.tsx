import React from "react";
import { ButtonVariant } from "@amzn/stencil-react-components/button";
import { Col } from "@amzn/stencil-react-components/layout";
import { Text } from "@amzn/stencil-react-components/text";
import { connect } from "react-redux";
import { ApplicationState } from "../../../../reducers/application.reducer";
import { FullBgcState } from "../../../../reducers/fullBgc.reducer";
import { JobState } from "../../../../reducers/job.reducer";
import { ScheduleState } from "../../../../reducers/schedule.reducer";
import { FullBgcStepConfig } from "../../../../utils/types/common";
import DebouncedButton from "../../../common/DebouncedButton";
import { goToNextFullBgcStep } from "../../../../utils/helper";
import { FULL_BGC_STEPS } from "../../../../utils/enums/common";

interface MapStateToProps {
  job: JobState;
  application: ApplicationState;
  schedule: ScheduleState;
  fullBgc: FullBgcState;
}

type ProofOfId1MergedProps = MapStateToProps;

export const ProofOfId1 = ( props: ProofOfId1MergedProps ) => {
  const { fullBgc } = props;
  const stepConfig = fullBgc.stepConfig as FullBgcStepConfig;

  const handleClickNext = () => {
    goToNextFullBgcStep(stepConfig, FULL_BGC_STEPS.PROOF_OF_ID1, FULL_BGC_STEPS.PROOF_OF_ID2);
  };

  return (
    <Col className="documentation-container" gridGap={15}>
      <Text fontSize="T200">
        ProofOfId1 Component
      </Text>
      ...
      <DebouncedButton variant={ButtonVariant.Secondary} onClick={handleClickNext}>
        Next
      </DebouncedButton>
    </Col>
  );
};

const mapStateToProps = ( state: MapStateToProps ) => {
  return state;
};

export default connect(mapStateToProps)(ProofOfId1);
