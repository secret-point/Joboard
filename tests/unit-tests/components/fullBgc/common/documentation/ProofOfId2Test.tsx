import React from "react";
import { shallow } from "enzyme";
import { ProofOfId2 } from "../../../../../../src/components/fullBgc/common/documentation/ProofOfId2";
import { TEST_APPLICATION_STATE, TEST_FULL_BGC_STATE, TEST_JOB_STATE, TEST_SCHEDULE_STATE } from "../../../../../test-utils/test-data";

describe("ProofOfId2", () => {

  it("should match snapshot", () => {
    const shallowWrapper = shallow(
      <ProofOfId2
        job={TEST_JOB_STATE}
        application={TEST_APPLICATION_STATE}
        schedule={TEST_SCHEDULE_STATE}
        fullBgc={TEST_FULL_BGC_STATE}
      />);

    expect(shallowWrapper).toMatchSnapshot();
  });
});
