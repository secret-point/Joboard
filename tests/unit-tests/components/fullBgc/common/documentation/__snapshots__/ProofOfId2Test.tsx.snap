// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProofOfId2 should match snapshot 1`] = `
<Col
  className="documentation-container"
  gridGap={15}
>
  <Text
    font="primary"
    fontSize="T200"
    hyphens="none"
  >
    ProofOfId2 Component
  </Text>
  ...
  <DebouncedButton
    onClick={[Function]}
    variant="secondary"
  >
    Next
  </DebouncedButton>
</Col>
`;
